### ISSUE TITLE
Code review

### ISSUE THREAD

**Issue by Smile4ever** [odebroqueville/contextSearch#1]
Hi Olivier,

As requested, some feedback on the code can be found below.

> if (info.menuItemId == 14) { // 14 is the index for LinkedIn

and
> searchEngines = sortAlphabetically(data);
>            var index = 0;
>            for (var se in searchEngines) {
>var strId = index.toString();

Please don't use menuItemId to check whether this is LinkedIn. This id can change at all times. Instead, use the key (or name property) from the JSON file as an id, that will work as well. 

Also note that due a bug with info.selectionText - https://bugzilla.mozilla.org/show_bug.cgi?id=1338898 - you probably should implement a workaround for that. Workaround in Translate Now:
> sendMessage("getSelection", selectionText, priviledgedSiteNoContentScript);

For all code on this matter, checkout https://github.com/Smile4ever/firefoxaddons/blob/master/Translate%20Now-webext/background.js and https://github.com/Smile4ever/firefoxaddons/blob/master/Translate%20Now-webext/translatenow.js

>  contexts: ["all"]

Please don't do this, and consider only those contexts that make sense. Currently you have only implemented "selection" (because when you right click the page without selection, nothing happens!)

>  var creating = browser.tabs.create({
>        url: targetUrl
>    });
>creating.then(null, onError);

I would shorten this to
>  browser.tabs.create({
>        url: targetUrl
>    }).then(null, onError);

or even

>  browser.tabs.create({ url: targetUrl });

Keep up the good work!

**Comment by odebroqueville** [comment]
Hi Geoffrey,
Thank you for your feedback.
I agree with your suggestions and amended the code. I had some difficulty getting my head around your advanced code to replace info.selectionText, so I used a content script that I named selection.js. I hadn't understood what contexts was about! Returning to the Mozilla documentation, I found a reference to ContextType which was what I was looking for.
I've pushed the code changes for you to verify before I submit version 3.13 to Firefox addons store. I don't think I've broken anything!
